src/azul/json.py

```diff
@@ -195,3 +197,23 @@ def json_head(n: int, o: AnyJSON) -> str:
         if buf.tell() > n:
             break
     return buf.getvalue()[:n]
+
+
+def json_hash(o: AnyJSON, hash_: _hashlib.HASH = None) -> _hashlib.HASH:
```

Comment:
```suggestion
def json_hash(o: AnyJSON, hash_: Optional[_hashlib.HASH] = None) -> _hashlib.HASH:
```

Either way, we're not supposed to use `_…` modules. However, I find the [prescribed](https://github.com/python/typeshed/issues/2928#issuecomment-487889925) alternative equally ugly and it doesn't work in PyCharm.  The best option for now is to omit the type hints.

Also, I've decided that we won't worry about local variables shadowing built-ins. Consider disabling that particular PyCharm warning.

So ultimately

```diff
Index: src/azul/json.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/json.py b/src/azul/json.py
--- a/src/azul/json.py	(revision 99dd040504c1a1070aa69479ec4558f55faab2d4)
+++ b/src/azul/json.py	(date 1697582214394)
@@ -1,4 +1,3 @@
-import _hashlib
 from copy import (
     copy,
     deepcopy,
@@ -199,21 +198,24 @@
     return buf.getvalue()[:n]
 
 
-def json_hash(o: AnyJSON, hash_: _hashlib.HASH = None) -> _hashlib.HASH:
+def json_hash(o: AnyJSON, hash=None):
     """
     Efficiently compute a hash of a JSON object.
 
     >>> o = {'foo': 1, 'bar': 2.0, 'baz': 'baz'}
+
     >>> json_hash(o).hexdigest()
     '08335acd02f77fdd32775f51a1766796e91bc0e1'
+
     >>> json_hash(o, hashlib.sha1()).hexdigest()
     '08335acd02f77fdd32775f51a1766796e91bc0e1'
+
     >>> json_hash(o, hashlib.md5()).hexdigest()
     'd28a433c1e34de7c7da3ea59fd9e48f9'
     """
-    if hash_ is None:
-        hash_ = hashlib.sha1()
+    if hash is None:
+        hash = hashlib.sha1()
     encoder = json.JSONEncoder(sort_keys=True, separators=(',', ':'))
     for chunk in encoder.iterencode(o):
-        hash_.update(chunk.encode())
-    return hash_
+        hash.update(chunk.encode())
+    return hash
```

Reply:
Warning disabled :+1: